src/main/kotlin/org/jitsi/nlj/rtp/codec/vp9/Vp9Packet.kt

```diff
@@ -0,0 +1,285 @@
+
+    val hasExtendedPictureId =
+        DePacketizer.VP9PayloadDescriptor.hasExtendedPictureId(buffer, payloadOffset, payloadLength)
+
+    val hasTL0PICIDX =
+        DePacketizer.VP9PayloadDescriptor.hasTL0PICIDX(buffer, payloadOffset, payloadLength)
+
+    val hasScalabilityStructure =
+        DePacketizer.VP9PayloadDescriptor.hasScalabilityStructure(buffer, payloadOffset, payloadLength)
+
+    val isUpperLevelReference =
+        DePacketizer.VP9PayloadDescriptor.isUpperLevelReference(buffer, payloadOffset, payloadLength)
+
+    val isInterPicturePredicted =
+        DePacketizer.VP9PayloadDescriptor.isInterPicturePredicted(buffer, payloadOffset, payloadLength)
+
+    private var _TL0PICIDX =
+        TL0PICIDX ?: DePacketizer.VP9PayloadDescriptor.getTL0PICIDX(buffer, payloadOffset, payloadLength)
+
+    var TL0PICIDX: Int
+        get() = _TL0PICIDX
+        set(newValue) {
+            _TL0PICIDX = newValue
+            if (newValue != -1 && !DePacketizer.VP9PayloadDescriptor.setTL0PICIDX(
+                    buffer, payloadOffset, payloadLength, newValue)) {
+                logger.cwarn { "Failed to set the TL0PICIDX of a VP9 packet." }
+            }
+        }
+
+    private var _pictureId =
+        pictureId ?: DePacketizer.VP9PayloadDescriptor.getPictureId(buffer, payloadOffset, payloadLength)
+
+    var pictureId: Int
+        get() = _pictureId
+        set(newValue) {
+            _pictureId = newValue
+            if (!DePacketizer.VP9PayloadDescriptor.setExtendedPictureId(
+                    buffer, payloadOffset, payloadLength, newValue)) {
+                logger.cwarn { "Failed to set the picture id of a VP9 packet." }
+            }
+        }
+
+    /* TODO: avoid recomputing these on clone */
+
+    val temporalLayerIndex: Int =
+        DePacketizer.VP9PayloadDescriptor.getTemporalLayerIndex(buffer, payloadOffset, payloadLength)
+
+    val spatialLayerIndex: Int =
+        DePacketizer.VP9PayloadDescriptor.getSpatialLayerIndex(buffer, payloadOffset, payloadLength)
+
+    val isSwitchingUpPoint: Boolean =
+        DePacketizer.VP9PayloadDescriptor.isSwitchingUpPoint(buffer, payloadOffset, payloadLength)
+
+    val usesInterLayerDependency: Boolean =
+        DePacketizer.VP9PayloadDescriptor.usesInterLayerDependency(buffer, payloadOffset, payloadLength)
+
+    fun getScalabilityStructure(
+        eid: Int = 0,
+        baseFrameRate: Double = 30.0
+    ): RtpEncodingDesc? {
+        var off = DePacketizer.VP9PayloadDescriptor.getScalabilityStructureOffset(buffer, payloadOffset, payloadLength)
+        if (off == -1) {
+            return null
+        }
+        val ssHeader = buffer[off].toInt()
+
+        val numSpatial = ((ssHeader and 0xE0) shr 5) + 1
+        val hasResolution = (ssHeader and Y_BIT) != 0
+        val hasPictureGroup = (ssHeader and G_BIT) != 0
+
+        off++
+
+        val heights = if (hasResolution) Array(numSpatial) { 0 } else null
+        if (hasResolution) {
```

Comment:
If I'm reading the below code right, you can fill `heights` with `RtpLayerDesc.NO_HEIGHT` if `hasResolution` is false and then avoid the `!!`s and `if (hasResolution)` checks.
So could be something like:
```
val heights = if (hasResolution) {
    Array(numSpatial) {
        off += 2 // We only record height, not width

        val height = ((buffer[off].toInt() and 0xff) shl 8) or (buffer[off + 1].toInt() and 0xff)
        off += 2
        height
   }
} else { 
    Array(numSpatial) { RtpLayerDesc.NO_HEIGHT } 
}
```